tests/unit/test_io_hdf5_h5tools.py

```diff
@@ -2958,6 +2958,27 @@ def test_append_data(self):
             self.assertEqual(f['foofile_data'].file.filename, self.paths[1])
             self.assertIsInstance(f.attrs['foo_ref_attr'], h5py.Reference)
 
+    def test_append_dataset_of_references(self):
+        """Test that exporting a written container with a dataset of references works."""
+        bazs = []
+        num_bazs = 10
+        for i in range(num_bazs):
+            bazs.append(Baz(name='baz%d' % i))
+        array_bazs=np.array(bazs)
+        wrapped_bazs = H5DataIO(array_bazs, maxshape=(None,))
+        baz_data = BazData(name='baz_data1', data=wrapped_bazs)
+        bucket = BazBucket(name='bucket1', bazs=bazs.copy(), baz_data=baz_data)
+
+        with HDF5IO(self.paths[0], manager=get_baz_buildmanager(), mode='w') as write_io:
+            write_io.write(bucket)
+
+        with HDF5IO(self.paths[0], manager=get_baz_buildmanager(), mode='a') as read_io:
+            read_bucket1 = read_io.read()
+            DoR = read_bucket1.baz_data.data
+            ref = read_bucket1.baz_data.data.dataset[0]
+            DoR.append(ref)
+            self.assertEqual(len(read_bucket1.baz_data.data), 11)
+
```

Comment:
Just in case, could you test closing this file, opening it, and making sure the size is still 11?

Reply:
The write *should* not be needed. Append should have acted directly on the file, just like changing the values of the dataset. But if it doesn't work, then that's odd and something we would need to look into or document.